File: ApiRest/IntermodularRest/src/main/java/es/intermodular/rest/modelo/HorarioCentro.java
package es.intermodular.rest.modelo;

import java.sql.Time;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
@Entity
@Table(name="horariocentro")
public class HorarioCentro {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	
	@Column(name = "horainicio", nullable=false)
	private Time horainicio;
	
	@Column(name = "horafin", nullable=false)
	private Time horafin;
	
	@Column(name = "turno", nullable=true)
	private Turno turno;

	//Getters y setters
	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public Time getHorainicio() {
		return horainicio;
	}

	public void setHorainicio(Time horainicio) {
		this.horainicio = horainicio;
	}

	public Time getHorafin() {
		return horafin;
	}

	public void setHorafin(Time horafin) {
		this.horafin = horafin;
	}

	public Turno getTurno() {
		return turno;
	}

	public void setTurno(Turno turno) {
		this.turno = turno;
	}
	
}

File: Guardiapp/app/src/main/java/com/intermodular/guardiapp/model/HorarioCentro.kt
package com.intermodular.guardiapp.model

import com.google.gson.annotations.SerializedName
import com.intermodular.guardiapp.viewmodel.Turno
import java.io.Serializable
import java.sql.Time

data class HorarioCentro(
    @SerializedName("id")
    private val id: Int,
    @SerializedName("horainicio")
    private val horainicio: Time,
    @SerializedName("horafin")
    private val horafin: Time,
    @SerializedName("turno")
    private val turno: Turno?
    ):Serializable
